src/overlays/actors/ovl_En_Ossan/z_en_ossan.c

```diff
@@ -37,227 +114,2406 @@ const ActorInit En_Ossan_InitVars = {
+    globalCtx->msgCtx.msgMode = 0x36;
+    globalCtx->msgCtx.unk_E3E7 = 4;
+    player->stateFlags2 &= ~0x20000000;
+    func_800BC490(globalCtx, 1);
+    Interface_ChangeAlpha(50);
+    this->drawCursor = 0;
+    EnOssan_UpdateCameraDirection(this, globalCtx, 0.0f);
+    this->stateFlag = ENOSSAN_STATE_GIVE_ITEM_FANFARE;
+    osSyncPrintf("\x1b[33m持ち上げ開始！！\x1b[m\n\n");
+}
+
+void EnOssan_SetStateCantGetItem(GlobalContext* globalCtx, EnOssan* this, u16 textId) {
+    func_8010B720(globalCtx, textId);
+    this->stateFlag = ENOSSAN_STATE_CANT_GET_ITEM;
+}
+
+void EnOssan_SetStateQuickBuyDialog(GlobalContext* globalCtx, EnOssan* this, u16 textId) {
+    func_8010B720(globalCtx, textId);
+    this->stateFlag = ENOSSAN_STATE_QUICK_BUY;
+}
+
+void EnOssan_HandleCanBuyItem(GlobalContext* globalCtx, EnOssan* this) {
+    EnGirlA* selectedItem = this->shelfSlots[this->cursorIndex];
+
+    switch (selectedItem->canBuyFunc(globalCtx, selectedItem)) {
+        case CANBUY_RESULT_SUCCESS_FANFARE:
+            if (selectedItem->actor.params == SI_HYLIAN_SHIELD && gSaveContext.infTable[7] & 0x40) {
+                EnOssan_SetStateGiveDiscountDialog(globalCtx, this);
+                break;
+            } else {
+                EnOssan_GiveItemWithFanfare(globalCtx, this);
+                this->drawCursor = 0;
+                this->shopItemSelectedTween = 0.0f;
+                selectedItem->setOutOfStockFunc(globalCtx, selectedItem);
+                return;
+            }
+        case CANBUY_RESULT_SUCCESS:
+            selectedItem->itemGiveFunc(globalCtx, selectedItem);
+            EnOssan_SetStateQuickBuyDialog(globalCtx, this, 0x84);
+            this->drawCursor = 0;
+            this->shopItemSelectedTween = 0.0f;
+            selectedItem->setOutOfStockFunc(globalCtx, selectedItem);
+            return;
+        case CANBUY_RESULT_CANT_GET_NOW:
+            func_80078884(NA_SE_SY_ERROR);
+            EnOssan_SetStateCantGetItem(globalCtx, this, 0x86);
+            return;
+        case CANBUY_RESULT_NEED_BOTTLE:
+            func_80078884(NA_SE_SY_ERROR);
+            EnOssan_SetStateCantGetItem(globalCtx, this, 0x96);
+            return;
+        case CANBUY_RESULT_NEED_RUPEES:
+            func_80078884(NA_SE_SY_ERROR);
+            EnOssan_SetStateCantGetItem(globalCtx, this, 0x85);
+            return;
+        case CANBUY_RESULT_CANT_GET_NOW_5:
+            func_80078884(NA_SE_SY_ERROR);
+            EnOssan_SetStateCantGetItem(globalCtx, this, 0x86);
+            return;
+    }
+}
+
+void EnOssan_HandleCanBuyLonLonMilk(GlobalContext* globalCtx, EnOssan* this) {
+    EnGirlA* item = this->shelfSlots[this->cursorIndex];
+
+    switch (item->canBuyFunc(globalCtx, item)) {
+        case CANBUY_RESULT_SUCCESS_FANFARE:
+            func_8010B720(globalCtx, 0x9C);
+            this->stateFlag = ENOSSAN_STATE_GIVE_LON_LON_MILK;
+            this->drawCursor = 0;
+            return;
```

Comment:
Same here about using breaks instead of returns, and I suggest including a break in the last case for consistency (relevant for a lot of other switches in this file as well)